app/src/main/java/at/aau/edu/wizards/gameModel/GameModelRules.kt

```diff
@@ -0,0 +1,211 @@
+package at.aau.edu.wizards.gameModel
+
+class GameModelRules(
+    private val players: ArrayList<GameModelPlayer>,
+    val id: Int,
+    private val cardDealer: GameModelDealer
+) {
+
+    val board = ArrayList<GameModelCard>()
+    private val wins = ArrayList<Int>()
+    var trump: GameModelCard = GameModelCard.NoCard
+        private set
+    private var round = 0
+    var currentPlayer = 0
+        private set
+    private var dealer = 0
+    private var winningPlayer = 0
+    var winningCard: GameModelCard = GameModelCard.NoCard
+
+    fun init() {
+        if (round == 0) {
+            round++
+            for (player in players) {
+                player.dealCards(round)
+            }
+            trump = cardDealer.dealCardInSet()
+            dealer = 0
+            currentPlayer = 0
+            cardDealer.resetSet()
+        }
+    }
+
+    private fun nextRound() {
+        if (round == 10) {
+            return endGame()
+        }
+        for (player in players) {
+            player.score(getAmountWon(player.id))
+        }
+        wins.clear()
+        round++
+        for (player in players) {
+            player.dealCards(round)
+        }
+        trump = cardDealer.dealCardInSet()
+        if (++dealer >= players.size) {
+            dealer = 0
+        }
+        currentPlayer = dealer
+        cardDealer.resetSet()
+    }
+
+
+    private fun endGame() {
+        //TODO
+    }
+
+    fun playCard(card: GameModelCard) {
+        if (card is GameModelCard.NoCard) {
```

Comment:
There are two reasons I decided to implement `NoCard`:

1: In wizards there is a special case where you try to deal over the card limit of 60 cards. Namely in the last round, when all cards are dealt between the players. If you then try to deal the card you would have no cards left, normally this would lead to an `ERROR`. Since the goal of this rework was to simplify the code I instead wrote the card dealer in a way that it returns a card of the instance `NoCard`, I can then go ahead and directly set this as the trump with the meaning: "There is no trump this round" instead of relying on more complex error handling to solve the issue.

2: I can avoid to use null-able variables in various parts of the code. Instead if there is no card for that variable at the moment you just set it to a card of instance `NoCard`. Avoiding any null casting, again improving readability.